e2e/airgap.go

```diff
@@ -5,19 +5,53 @@ import (
 	"fmt"
 	"log"
 	"net"
+	"strings"
 
 	"github.com/Azure/agentbakere2e/suite"
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
 	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork"
 )
 
 var (
-	nsgName           string = "abe2e-airgap-securityGroup"
+	airgapNSGName     string = "abe2e-airgap-securityGroup"
 	defaultSubnetName string = "aks-subnet"
 )
 
-func airGapSecurityGroup(location, kubernetesEndpont string) armnetwork.SecurityGroup {
-	allowVnet := armnetwork.SecurityRule{
+func getRequiredSecurityRules(clusterFQDN string) ([]*armnetwork.SecurityRule, error) {
+	// https://learn.microsoft.com/en-us/azure/aks/outbound-rules-control-egress#azure-global-required-fqdn--application-rules
+	// note that we explicitly exclude packages.microsoft.com
+	requiredDNSNames := []string{
+		"mcr.microsoft.com",
+		"management.azure.com",
+		"acs-mirror.azureedge.net",
+		clusterFQDN,
+	}
+	var rules []*armnetwork.SecurityRule
+	var priority int32 = 100
```

Comment:
could we call this startingPriority or initialPrioirty ?

Reply:
personally I think it makes sense as is since it gets incremented and applied to every rule